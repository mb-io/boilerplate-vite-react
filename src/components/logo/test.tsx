import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import Logo from './logo'

// Mock the useThemeStore hook
vi.mock('@/stores/theme-store/theme-store', () => ({
  default: () => ({
    theme: 'light' // default theme for testing
  })
}))

describe('Logo Component', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<Logo />)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly with size "sm"', () => {
    const { container } = render(<Logo size="sm" />)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly with size "md"', () => {
    const { container } = render(<Logo size="md" />)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly with size "lg"', () => {
    const { container } = render(<Logo size="lg" />)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly with custom className', () => {
    // eslint-disable-next-line tailwindcss/no-custom-classname
    const { container } = render(<Logo className="custom-class" />)
    expect(container).toMatchSnapshot()
  })
})
