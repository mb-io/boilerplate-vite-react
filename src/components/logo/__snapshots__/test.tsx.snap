// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo Component > renders correctly with custom className 1`] = `
<div>
  <picture>
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.webp"
      type="image/webp"
    />
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.png"
      type="image/png"
    />
    <img
      alt="MeteorByte Logo"
      class="custom-class"
      height="52px"
      src="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.png"
      width="200px"
    />
  </picture>
</div>
`;

exports[`Logo Component > renders correctly with default props 1`] = `
<div>
  <picture>
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.webp"
      type="image/webp"
    />
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.png"
      type="image/png"
    />
    <img
      alt="MeteorByte Logo"
      height="52px"
      src="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.png"
      width="200px"
    />
  </picture>
</div>
`;

exports[`Logo Component > renders correctly with size "lg" 1`] = `
<div>
  <picture>
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-400px.webp"
      type="image/webp"
    />
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-400px.png"
      type="image/png"
    />
    <img
      alt="MeteorByte Logo"
      height="104px"
      src="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-400px.png"
      width="400px"
    />
  </picture>
</div>
`;

exports[`Logo Component > renders correctly with size "md" 1`] = `
<div>
  <picture>
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-300px.webp"
      type="image/webp"
    />
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-300px.png"
      type="image/png"
    />
    <img
      alt="MeteorByte Logo"
      height="78px"
      src="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-300px.png"
      width="300px"
    />
  </picture>
</div>
`;

exports[`Logo Component > renders correctly with size "sm" 1`] = `
<div>
  <picture>
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.webp"
      type="image/webp"
    />
    <source
      srcset="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.png"
      type="image/png"
    />
    <img
      alt="MeteorByte Logo"
      height="52px"
      src="https://meteorbytemediablob.blob.core.windows.net/public-media/MeteorByte-Logo-Light-200px.png"
      width="200px"
    />
  </picture>
</div>
`;
